### ISSUE TITLE
Plugin not working when moment-timezone is not in 'node_modules' directory

### ISSUE THREAD

**Issue by diniciacci** [gilmoreorless/moment-timezone-data-webpack-plugin#32]
Currently the filter does not work if moment-timezone is located in a different directory than 'node_modules'.

This is due to [line 199](https://github.com/gilmoreorless/moment-timezone-data-webpack-plugin/blob/1125735c75d5a5cd27c214f4db28fbf0c31e7bbd/src/index.js#L199) where 'node_modules' is hard coded in the path to match the context.

Having modules located in other directories is supported by webpack as [version 5](https://webpack.js.org/configuration/resolve/#resolvemodules) and also [version 4](https://v4.webpack.js.org/configuration/resolve/#resolvemodules).

IMHO a fix could be to just match '/moment-timezone' in the context.

On the contrary a more extended fix could involve matching all the values of resolve.modules array + 'moment-timezone'.

I can provide a PR if you want.

**Comment by gilmoreorless** [comment]
Good catch! I've used `resolve.modules` in some projects before, but not with moment-timezone, so I hadn't come across this problem.

For your use case, when you have moment-timezone in a different directory, does it still have the same file/directory structure as it would in `node_modules`? I just want to make sure that `require('moment-timezone/data/packed/latest.json')` would still work as expected.

**Comment by diniciacci** [comment]
Thanks ;)

My use case comes from simply porting a legacy code to webpack and, as not so uncommon, the libraries for the client are located in a "vendor"  directory. In particular this is a dojo toolkit project, that had its own (very good) build system that stored libs in arbitrary (configurable) paths.

To sum it up, yes in my case moment-timezone was just located in "vendor" instead of "node_modules". In general for dojo that might not be the case.

Notice that from my few tests for my use case, there is no way to make the plugin work without editing its code. This is because to me it looks like webpack will always resolve the context to the actual absolute path (e.g. even if you create symlinks or try to circumvent the loading location with aliases).

Imho also fixing 'moment-timezone/data/packed/latest.json' scares me a bit and maybe should be a configurable value? timezoneDataFilename? Nonetheless in my usecase the data filename was not a problem.

Let me know if I can be of any help, I'm currently hunting another one around here ;)

**Comment by gilmoreorless** [comment]
That's good context, thanks.

> Imho also fixing 'moment-timezone/data/packed/latest.json' scares me a bit and maybe should be a configurable value? timezoneDataFilename?

Yep I have wondered in the past about that hard-coded reference, but I left it alone until an option for it was actually needed. It looks like you've found that use case.

I'm a bit pressed for time at the moment, so if you're happy to put up a PR about this, it would be super-helpful.

**Comment by diniciacci** [comment]
Hey,

the PR is ready, but I'm stuck on the test. Would like to provide something more than a unit test, cause the modification albeit small has quite some use consequences. I.e. I need a brilliant idea on how to test it.

**Comment by gilmoreorless** [comment]
Ah yes, that's an interesting question. Some of the tests already use [`memory-fs`](https://www.npmjs.com/package/memory-fs) for checking the webpack-compiled output. Perhaps you could create a new `MemoryFS` as the input filesystem and write whatever temporary files you need to it? (I haven't actually tried this myself.)

**Comment by diniciacci** [comment]
Thanks, this is exactly the kind of suggestion I was looking for. I will try this way and see if I can get it running.

**Comment by diniciacci** [comment]
I did the pull request as the code was getting stale. The reason for the wait is that there is one missing test that I really wanted for the code.

I left the missing test string commented in case someone gets a good idea on how to code it properly. My only idea would have meant heavily touching the current test harnesses and looked like an overkill.

I didn't continue on the memory-fs path because it is deprecated and getting even more dependent on it sounded bad to me. At the same time moving everything to [memfs](https://github.com/streamich/memfs) looked a bit out of scope of the fix.

**Comment by gilmoreorless** [comment]
I took a quick look at that commented-out test. You're right, it would be best to switch to `memfs` (and [`linkfs`](https://github.com/streamich/linkfs)) to implement it properly. I've created #36 to tackle that separately.

I've released your PR as version `1.4.0` so that it's not held up by the test fixes.

**Comment by diniciacci** [comment]
Thanks. I really agree with you. I totally missed linkfs and it is indeed very interesting. I'll have a look too.
